app/tests/e2e.transformer.test.ts

```diff
@@ -0,0 +1,132 @@
+const testTransformer = "Test transformer to be deleted";
+
+describe("Transformer API tests", () => {
+  it("should create a transformer", async () => {
+    expect.assertions(3);
+
+    // When a new transformer is posted
+    const response = await fetch(tansformerBaseUrl, {
+      method: "post",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        name: testTransformer,
+        transformer: `(body, req, res) => {
+              res.statusCode = 418;
+              const json = JSON.parse(body);
+              json.message = "Newly transformed";
+              return JSON.stringify(json);
+          }`,
+      }),
+    });
+
+    const message = await response.json();
+
+    // Then expect the newly posted transformer to be returned with a status code of 201
+    expect(response.status).toBe(201);
+    expect(message.name).toBe(transformerCreateExpected.name);
+    expect(message.source.replace(/\s/g, "")).toBe(
+      transformerCreateExpected.source.replace(/\s/g, "")
+    );
+  });
+
+  it("should get the transformers", async () => {
+    expect.assertions(5);
+
+    // When transformers are requested
+    const response = await fetch(tansformerBaseUrl, {
+      method: "get",
+    });
+
+    const message = await response.json();
+    /**
+     * .length call does not work on imported json modules.
+     * Imported modules a key for each member of the array, and a default key.
+     * Hence counting the number of keys and then subtracting by one.
+     */
+    const expectedLength = Object.keys(transformerGetExpected).length - 1;
+
+    // Then expect all transformers to be returned
+    expect(message.length).toBe(expectedLength);
+    expect(message[0].name).toBe(transformerGetExpected[0].name);
+    expect(message[1].name).toBe(transformerGetExpected[1].name);
+    expect(message[1].source.replace(/\s/g, "")).toBe(
+      transformerGetExpected[1].source.replace(/\s/g, "")
+    );
+    expect(response.status).toBe(200);
+  });
+
+  it("should get the transformed message with transformed status code", async () => {
+    expect.assertions(2);
+
+    // When a get request is sent after creating a new transformer
+    const response = await fetch(baseUrl, {
+      method: "get",
+    });
+
+    const message = await response.json();
+
+    // Then expect the new transformer to modify the message and status code
+    expect(response.status).toBe(418);
+    expect(message).toEqual({ message: "Newly transformed" });
+  });
+
+  it("should get a transformer by id and delete it", async () => {
+    expect.assertions(4);
+
+    // Given there are existing transformers
+    const response = await fetch(tansformerBaseUrl, {
+      method: "get",
+    });
+
+    expect(response.status).toBe(200);
+
+    // Get the id of the transformer to be deleted
+    const message = await response.json();
+    const toDeleteId = message.filter(
+      (element: any) => element.name === testTransformer
```

Comment:
Can a better type be defined here instead of an `any`

Reply:
Used `{ id: string; name: string; source: string }`